### PR TITLE
Offcanvas optional `hide()` when resizing the window

### PR DESCRIPTION
**Pull request by ghost**
### Description

When the user resizes the window, all offcanvas that are not in the `fixed` position are hidden. This pull request is intended to make this behavior optional with an option called `hideNotFixedOnWindowResizing` (`true` by default). It fixes https://github.com/twbs/bootstrap/issues/37916.

### Motivation & Context

In some cases we don't want to hide the offcanvas on window resizing, even when its position is absolute.

Scenario example:

1. The user uses an Android mobile device.
2. The DOM has an `.offcanvas` with `position: absolute`.
3. The `.offcanvas` contains an `<input type="text">`.
4. The user click on the `<input type="text">` to write something.
5. The `.offcanvas`  will be hidden because its position is not `fixed`. Explanation:
    - Clicking on the `<input type="text">` will open the mobile virtual keyboard.
    - When opening the mobile virtual keyboard, the `resize` event will be triggered on the `window`.
    - Because of this event, the current listener will be executed.

### Type of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would change existing functionality)

### Checklist

- [x] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)
- [x] My code follows the code style of the project _(using `npm run lint`)_
- [x] My change introduces changes to the documentation
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

#### Live previews

- <https://deploy-preview-38125--twbs-bootstrap.netlify.app/>

### Related issues

Fixes #37916

### CODE REVIEW

File: js/src/offcanvas.js [hunk 1]
```diff
@@ -266,7 +268,10 @@ EventHandler.on(window, EVENT_LOAD_DATA_API, () => {
 EventHandler.on(window, EVENT_RESIZE, () => {
   for (const element of SelectorEngine.find('[aria-modal][class*=show][class*=offcanvas-]')) {
     if (getComputedStyle(element).position !== 'fixed') {
-      Offcanvas.getOrCreateInstance(element).hide()
+      const data = Offcanvas.getOrCreateInstance(element)
+      if (data._config.hideNotFixedOnWindowResizing) {
```
Comment (XhmikosR) [inline js/src/offcanvas.js:272]: Why are you using the `_config` here? Isn't there another property that doesn't start with an underscore to use?

In the future we might mangle all properties that start with an underscore so this is a bad practice (let alone that it's a common convention for private stuff).

Comment (XhmikosR) [reply js/src/offcanvas.js:272]: Hmm, indeed it might be OK because we inherit from the base component.

We might want to revisit this later @GeoSot :)

**Comment by XhmikosR** [comment]
@GeoSot could you please have a look at the PR?

File: js/src/offcanvas.js (hunk 1 above)
Comment (GeoSot) [reply js/src/offcanvas.js:272]: As you already wrote, and I am pretty sure, I have commented in another similar approach, we should avoid referring to class 'private' properties.
It of course work on this code, but even if we mangle the properties or use native [private members](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields), we will not be able to replicate this behavior

Comment (XhmikosR) [reply js/src/offcanvas.js:272]: The thing is that we currently use this everywhere and this PR just follows the same scheme.

How else can this be done? And we should probably try to do the same across the whole codebase in a separate PR

**Comment by XhmikosR** [comment]
@GeoSot can you please check this PR?

**Pull request closed without merging**